### DISCUSSION TITLE
Add a vscode extension that is not listed in the open VSX registry

### DISCUSSION THREAD

**Discussion by cmakasy** [eclipse-theia/theia#14032] [Q&A]
I would like to add the following vs code extension eclipe-sirius.sirius-web to theia ide. But  unfortunately this extension is not available in the open VSX registry. Does anybody an idea how can I add this extension.

Many thanks in advance

**Comment by YathindraNikshipth** [comment]
You can download the VSIX file from the VSCode Marketplace, extract its contents into a folder, and then copy that folder into the plugins folder of your Theia application. I followed this process, and it worked.

**Reply by cmakasy** [reply]
Many thanks for your response, but I don't see the option download vsix file. (I added the extensions to VSCode, no problem). Now  I want to integrate that extensions into da dockerfile but I can't find something like this "vscode-eslint": "https://github.com/theia-ide/vscode-eslint/releases/download/release%2F2.0.15/vscode-eslint-2.0.15.vsix" for the sirius-web extension. Could you please give me a hint/link where I could find this / how to do this. Many thanks in advance

**Reply by msujew** [reply]
@cmakasy You can find the download link (https://marketplace.visualstudio.com/_apis/public/gallery/publishers/eclipse-sirius/vsextensions/sirius-web/2022.9.6/vspackage) for the extension on the extension page:

![image](https://github.com/user-attachments/assets/98df53f7-565f-4685-bdd0-a0eaf9ed256b)

Please note that manually downloading extensions and installing them into Theia is against the [Terms of Use](https://cdn.vsassets.io/v/M190_20210811.1/_content/Microsoft-Visual-Studio-Marketplace-Terms-of-Use.pdf) of the VS Code marketplace. This is the reason why open-vsx exists in the first place. I've raised an issue with the sirius team for this https://github.com/eclipse-sirius/sirius-web/issues/3853.

**Reply by cmakasy** [reply]
Thanks

**Comment by msujew** [comment]
Hey @cmakasy,

You can use the `Install from VSIX` command available in the extension view to install any `vsix` file in your file system:

![image](https://github.com/user-attachments/assets/4131fe6a-3a5a-4f5d-8272-ac84949a6221)

Additionally, given that it's an Eclipse project, I can imagine the people over at the Sirius project are open to publishing the extension on open-vsx as well. I've raised an issue for this over at https://github.com/eclipse-sirius/sirius-web/issues/3853.

Please note that manually downloading extensions and installing them into Theia is against the [Terms of Use](https://cdn.vsassets.io/v/M190_20210811.1/_content/Microsoft-Visual-Studio-Marketplace-Terms-of-Use.pdf) of the VS Code marketplace. This is the reason why open-vsx exists in the first place.
